pkg/rke2/rke2.go

```diff
@@ -166,5 +176,151 @@ func setup(clx *cli.Context, cfg Config) error {
 	}
 	executor.Set(&sp)
 
+	disabledItems := map[string]bool{
+		"kube-apiserver":          disableAPIServer,
+		"kube-scheduler":          disableScheduler,
+		"kube-controller-manager": disableControllerManager,
+		"etcd":                    disableETCD,
+	}
+	return removeOldPodManifests(dataDir, disabledItems)
+}
+
+func podManifestsDir(dataDir string) string {
+	return filepath.Join(dataDir, "agent", defaultPodManifestPath)
+}
+
+func removeOldPodManifests(dataDir string, disabledItems map[string]bool) error {
+	var kubeletStandAlone bool
+
+	kubeletErr := make(chan error)
+	containerdErr := make(chan error)
+
+	ctx := context.Background()
+
+	for component, disabled := range disabledItems {
+		if disabled {
+			manifestName := filepath.Join(podManifestsDir(dataDir), component+".yaml")
```

Comment:
Don't call `podManfestsDir` inside the loop
```suggestion
	agentManifestsDir := podManifestsDir(dataDir)
	for component, disabled := range disabledItems {
		if disabled {
			manifestName := filepath.Join(agentManifestsDir, component+".yaml")
```